### ISSUE TITLE
common_jit.check_against_reference() should check gradients w.r.t. weights

### ISSUE THREAD

**Issue by dagitses** [pytorch/pytorch#52052]
## 🚀 Feature
`common_jit.check_against_reference()` should check gradients w.r.t. weights and automatically check gradients w.r.t. inputs if any inputs require them. Then we can remove the need for requesting gradient checking (it will happen automatically).

## Motivation

We should check gradients w.r.t. weights because it will improve the accuracy of the comparison to the reference implementation.
Removing the need to request gradients explicitly simplifies the authoring of tests.

## Pitch

1. Augment `check_against_reference()` to check gradients wrt weights.
2. Check gradients wrt inputs only if inputs require them.
3. Remove `no_grad` parameter.

## Alternatives

We could keep the `no_grad` parameter if we want to keep it as the test author indicating that they expect gradients to be present. In which case, we would make it an error if gradients are expected but not requested. Renaming it to `expect_input_gradients` would make sense.

## Additional context

See https://github.com/pytorch/pytorch/pull/52016.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
